snmp/datadog_checks/snmp/config.py

```diff
@@ -79,7 +79,7 @@ def __init__(
             if value in (None, ""):
                 instance.pop(key)
 
-        self.logger = weakref.ref(local_logger) if logger is None else weakref.ref(logger)
+        self.get_logger = weakref.ref(local_logger) if logger is None else weakref.ref(logger)
```

Comment:
note: since it's a callable, let's call it `get_logger` to make it more explicit it should be called.